### ISSUE TITLE
Add an initial implementation of the Arcade official templates, compatible with 1ES pipeline templates.

### ISSUE THREAD

**Issue by riarenas** [dotnet/arcade#14524]
- [ ] This issue is blocking 
- [ ] This issue is causing unreasonable pain

We are going to introduce a new set of templates, under eng/common/templates-official that will be compatible with 1ES pipeline templates. We can keep refining these templates to remove public functionality, refactor common things, and make performance optimizations to outputs while the product repos are still working on their migration.

**Comment by riarenas** [comment]
We need this in every branch that has ever existed. Back to dev.

**Comment by MichaelSimons** [comment]
Nit - the title of this issue gives the impression it can be closed once the initial implementation is complete.  The second line in the description is very different however.

> We can keep refining these templates to remove public functionality, refactor common things, and make performance optimizations to outputs while the product repos are still working on their migration.

Please consider updating the title to be more broad or breaking out the cleanup/optimization work into a separate issue.

**Comment by riarenas** [comment]
Still working on getting templates out for servicing branches. Has been difficult to test as the simple repos I've been using for testing do not have servicing branches.

**Comment by riarenas** [comment]
Templates for all branches are now available for repos
